Fakebook.Posts/Fakebook.Posts.RestApi/Controllers/PostsController.cs

```diff
@@ -233,5 +239,27 @@ public async Task<IActionResult> GetNewsfeedAsync()
             return NoContent();
         }
 
+        [HttpPost("UploadPicture"), DisableRequestSizeLimit]
+        public async Task<ActionResult> UploadPicture() 
+        {
+            try {
+                if (Request.Form.Files[0] is IFormFile file)
+                {
+                    // generate a random guid from the file name
+                    var newFileName = $"{Request.Form["userId"]}-{Guid.NewGuid()}.{file.FileName.Split('.').Last()}";
+                    // upload image
+                    var result = await _blobService.UploadFileBlobAsync(
+                        "fakebook",
+                        file.OpenReadStream(),
+                        file.ContentType,
+                        newFileName);
+                    return Ok(new { path = result.AbsoluteUri });
+                } else return BadRequest();
+            } catch (Exception ex) {
+                _logger.LogError(ex, ex.Message);
+                return BadRequest(ex.Message);
```

Comment:
arbitrary exceptions should not be caught and turned into bad request. only things that you can concretely attribute to the request should be bad request, the rest just don't catch them and let asp.net return 500. anyway some exceptions might have details in their message that shouldn't be told to the client for security.

Reply:
Specified exceptions for invalid operation, request failed and a 500 for bad files